src/workerd/api/basics.c++

```diff
@@ -536,27 +544,36 @@ class AbortTriggerRpcClient final {
   }
 
   ~AbortTriggerRpcClient() noexcept(false) {
-    if (skipReleaseForTest) {
-      return;
-    }
-
-    auto req = client.releaseRequest(capnp::MessageSize{4, 0});
-    // We call detach() on the resulting promise so that we can perform RPC in a destructor
-    req.sendIgnoringResult().detach([](kj::Exception exc) {
-      if (exc.getType() == kj::Exception::Type::DISCONNECTED) {
-        // It's possible we can't send the release message because we're already disconnected.
-        return;
-      };
-
-      // Other exceptions could be more interesting
-      LOG_EXCEPTION("abortTriggerReleaseRpc", exc);
-    });
+    releaseFulfiller->fulfill(Release{kj::mv(client), skipReleaseForTest});
   }
 
   bool skipReleaseForTest = false;
 
+  static IoOwn<AbortTriggerRpcClient> create(IoContext& ioctx, rpc::AbortTrigger::Client&& client) {
+    auto [promise, fulfiller] = kj::newPromiseAndFulfiller<Release>();
+    ioctx.addTask(promise.then([](Release release) -> kj::Promise<void> {
```

Comment:
Hmm, on one hand this strategy fixes a problem @npaun observed where, at IoContext destruction, the CustomEvent for the outgoing RPC session might be torn down before the AbortTrigger, causing the release call to fail.

On the other hand, though, I  think this `addTask` effectively means that after a Durable Object has sent an AbortSignal somewhere, it may be blocked from hibernating while it waits for the AbortTrigger to be GC'd? Or... maybe that problem exists either way, because the CustomEvent itself is registered as a task, and the AbortTrigger holds open the RPC session even after everything else has finished?

Reply:
Is this blocking or should I continue waiting for a review from @npaun?